assets/js/hooks/js_view/iframe.js

```diff
@@ -28,8 +28,8 @@ import { sha256Base64 } from "../../lib/utils";
 
 const IFRAME_SHA256 = "fA00WeO9LAvgpbMz9vKEU0WTr4Uk5bTt/BxKHdweEz8=";
 
-export function initializeIframeSource(iframe, iframePort) {
-  const iframeUrl = getIframeUrl(iframePort);
+export function initializeIframeSource(iframe, iframePort, iframeUrl) {
+  if (!iframeUrl) iframeUrl = getIframeUrl(iframePort);
```

Comment:
Let's move this logic to `getIframeUrl`, and we should rewrite the URL protocol to match window.location.

Reply:
I'm not sure what you mean by "Rewrite the URL protocol to match window.location".   Here's what I'm thinking about the refactored `getIframeUrl` function:

```javascript
function getIframeUrl({iframePort: iframePort, iframeUrl: iframeUrl}) {
  if (iframeUrl) return iframeUrl;

  return window.location.protocol === "https:"
    ? "https://livebook.space/iframe/v3.html"
    : `http://${window.location.hostname}:${iframePort}/iframe/v3.html`;
}
```

Reply:
I'm thinking something like this:

```js
function getIframeUrl(iframePort, iframeUrl) {
  const protocol = window.location.protocol;

  if (iframeUrl) {
    return iframeUrl.replace(/^https?:/, protocol);
  }

  return protocol === "https:"
    ? "https://livebook.space/iframe/v3.html"
    : `http://${window.location.hostname}:${iframePort}/iframe/v3.html`;
}
```

In practice when someone specifies custom URL they will be accessing Livebook over https://, but in general we need to load using the same protocol as Livebook itself, so for completeness I'd override the URL scheme :)

Reply:
Cool, gonna make those changes, test it out, then will push up!